### PR TITLE
Colorize genetic models on variantS page

### LINKED ISSUE #3631: One colour per inheritance pattern?
It would be nice to show different inheritance patterns in different colours
In this view 
image
and in this view
image
to make sure one doesn´t misstake for example a hom variant for a het.

### PR DESCRIPTION
**Pull request by northwestwitch**
Second partial fix #3631. **TO BE REVIEWED PREFERABLY AFTER #3664**, because contains code from it.

Changes RD variantS table from this:

image

To this:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by

### CODE REVIEW

File: scout/constants/variant_tags.py [hunk 1]
```diff
@@ -96,6 +98,20 @@
     ("XD_dn", "X Linked Dominant De Novo"),
 )
 
+# Associates an inheritance model with a color, using the same colors categories as scout.constants.gene_tags.INHERITANCE_PALETTE
+# The colors of this palette are the same as the MatPlotLib Tableau palette (https://matplotlib.org/3.4.1/gallery/color/named_colors.html)
+GENETIC_MODELS_PALETTE = {
+    "AR_hom": INHERITANCE_PALETTE["AR"],
```
Comment (dnil) [inline scout/constants/variant_tags.py:104]: Here I think we will want a graded intensity. AR should be more visible than AR_comp (which anyway has a Compund button connected, and will be nice with a thematic color), AD_dn more intense than AD etc.

File: scout/constants/variant_tags.py [hunk 2]
```diff
@@ -96,6 +98,20 @@
     ("XD_dn", "X Linked Dominant De Novo"),
 )
 
+# Associates an inheritance model with a color, using the same colors categories as scout.constants.gene_tags.INHERITANCE_PALETTE
+# The colors of this palette are the same as the MatPlotLib Tableau palette (https://matplotlib.org/3.4.1/gallery/color/named_colors.html)
+GENETIC_MODELS_PALETTE = {
```
Comment (dnil) [inline scout/constants/variant_tags.py:103]: 👍 Good with the same base colors!

**Review by dnil — CHANGES_REQUESTED** [review]

**Comment by northwestwitch** [comment]
I'm working on the graded intensity and I've found a solution using opacity. Works nicely on light mode:

image


But it doesn't render OK on dark mode: 😢 

image

**Comment by northwestwitch** [comment]
Ok, I've modified the genetic models (variantS oage) like this:

image

And the inheritance models (variant page) like this:

image

**Review by dnil — APPROVED** [review]
Looks good! If you want, I could have a look at your opacity fix - maybe I could coerce the dark mode - but otherwise it is ok like this, isn't it?

**Comment by northwestwitch** [comment]
> Looks good! If you want, I could have a look at your opacity fix - maybe I could coerce the dark mode - but otherwise it is ok like this, isn't it?

The opacity fix looks really nice on the light theme. I have tried to force the badges to stay in light mode also if the system is in dark mode with this css style: `color-scheme: only light;` but didn't work. If you have time yes, give it a try (on another branch perhaps?) You can pull one of the intermediate commits of this branch in the new branch and test. Thanks!

**Comment by dnil** [comment]
> > Looks good! If you want, I could have a look at your opacity fix - maybe I could coerce the dark mode - but otherwise it is ok like this, isn't it?
> 
> The opacity fix looks really nice on the light theme. I have tried to force the badges to stay in light mode also if the system is in dark mode with this css style: `color-scheme: only light;` but didn't work. If you have time yes, give it a try (on another branch perhaps?) You can pull one of the intermediate commits of this branch in the new branch and test. Thanks!

Hm, you may only have done this on your local? Couldn't find the commit? Anyway, its nice enough this way so agreed, new branch and at QOL priority. Meeting ahead...

**Comment by northwestwitch** [comment]
> Hm, you may only have done this on your local?

Perhaps! A disadvantage of **opacity** is that the default style applied to badge color but also badge text. 

I remember using the --bs-bg-opacity style: https://getbootstrap.com/docs/5.2/utilities/background/#opacity

**Comment by dnil** [comment]
> > Hm, you may only have done this on your local?
> 
> Perhaps! A disadvantage of **opacity** is that the default style is applied to badge color but also badge text. Example: https://getbootstrap.com/docs/5.1/utilities/opacity/
> 
> I remember using the `bg-opacity-x` style: https://getbootstrap.com/docs/5.2/utilities/background/#opacity

Looks super sweet! Let's see, aren't we/darkmode still on 5.0? Perhaps we should update..

**Comment by dnil** [comment]
> > Perhaps! A disadvantage of **opacity** is that the default style is applied to badge color but also badge text. Example: https://getbootstrap.com/docs/5.1/utilities/opacity/
> > I remember using the `bg-opacity-x` style: https://getbootstrap.com/docs/5.2/utilities/background/#opacity
> 
> Looks super sweet! Let's see, aren't we/darkmode still on 5.0? Perhaps we should update..

Yeah, well, bootstrap dark stopped main at 5.1.3, since the 5.2-beta color changes were rather big. There is now also a less intrusive dark mode version with only the deltas, not the whole base css:  bootstrap-dark-plugin, but I think its also just at 5.1.3. 

It seems right now they will not fix the core issue with 5.2 in the 5.x, delaying it for v6 https://github.com/twbs/bootstrap/issues/36458. Hmm, maybe absolute colors is not so bad... 😆

**Comment by dnil** [comment]
> It seems right now they will not fix the core issue with 5.2 in the 5.x, delaying it for v6 [twbs/bootstrap#36458](https://github.com/twbs/bootstrap/issues/36458). Hmm, maybe absolute colors is not so bad... 😆

Ah, actually he does have a dev branch for 5.2: https://github.com/vinorodrigues/bootstrap-dark-5/tree/dev/v1.2.0
So we could do that, but...

**Pull request merged**